clib/dpd.js

```diff
@@ -57,9 +56,15 @@
         root += ':' + location.port;
       }
     }
-
+    if (root !== oldRoot) {
+      if (socket && socket.io) {
+        // disconnect socket if we're changing url
+        socket.io.disconnect();
+      }
+      socket = null;
```

Comment:
Oh, may need to warn dev that when they change base url, they may need to re-listen socket events